modules/appeals_api/app/services/concerns/appeals_api/stamp_pdf.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require 'central_mail/datestamp_pdf'
+
+module AppealsApi
+  module StampPdf
+    extend ActiveSupport::Concern
+
+    # rubocop:disable Metrics/BlockLength
+    included do
+      def stamp
+        if v1_hlr?
+          structure.stamp(consumer_stamp_path)
+        else
+          CentralMail::DatestampPdf.new(nil).stamp(time_stamp_path, name_and_social_stamp_path)
+        end
+      end
+
+      def consumer_stamp_path
+        CentralMail::DatestampPdf.new(@unstamped_path).run(
+          text: "Submitted by #{appeal.consumer_name} via api.va.gov",
+          x: 429,
+          y: 775,
+          text_only: true
+        )
+      end
+
+      def time_stamp_path
+        return if v1_hlr?
+
+        CentralMail::DatestampPdf.new(consumer_stamp_path).run(
+          text: "API.VA.GOV #{Time.zone.now.utc.strftime('%Y-%m-%d %H:%M%Z')}",
+          x: 5,
+          y: 775,
+          text_only: true
+        )
+      end
+
+      def name_and_social_stamp_path
+        return if v1_hlr?
+
+        stamp_path = "#{Common::FileHelpers.random_file_path}.pdf"
+        Prawn::Document.generate(stamp_path, margin: [0, 0]) do |pdf|
+          pdf.text_box stamp_text(appeal.veteran_last_name, appeal.ssn),
+                       at: [205, 785],
+                       align: :center,
+                       valign: :center,
+                       overflow: :shrink_to_fit,
+                       min_font_size: 8,
+                       width: 215,
+                       height: 10
+        end
+        stamp_path
+      end
+
+      private
+
+      def stamp_text(veteran_last_name, veteran_ssn)
+        "#{veteran_last_name.truncate(35)} - #{veteran_ssn.last(4)}"
+      end
+
+      def v1_hlr?
+        # TODO: remove this method and the conditional logic reliant upon it once V1 HLR goes bye bye
```

Comment:
It could be a year or more before we actually remove the first version of HLR.. IMO this is a pretty tight dependency on a specific version of a specific form, rather than a generic stamping module. 

Can you help me understand what problem this refactor is solving?

Reply:
We can't stamp the the v1 hlr with last name and social, like we're stamping nod and v2 hlr.

Reply:
It's generic for hlr v2, nod v1, and eventually for sc

Reply:
I'm trying to understand how the introduction of this pattern solves that issue

Reply:
I really don't mind distributing all of this in the individual appeal structures, but there was a comment in the code stating that upon v2 hlr, the expectation was to refactor stamping to be type-of-appeal independent, so I went down that road. I can rewind and go down the other (each pdf structure holds onto it's own stamp_text) if everyone agrees.

Reply:
As it stands right now, the Structure file of each form is what contains the unique business logic of that specific form. I think it might make sense to include this module in the structure file rather than the generator, and just not call the stamp method from the module inside v1 hlr, instead of doing the conditional logic inside the module itself.

Reply:
Got it.